src/IO/ReadHelpers.cpp

```diff
@@ -508,10 +508,32 @@ void readEscapedStringInto(Vector & s, ReadBuffer & buf)
             return;
 
         if (*buf.position() == '\\')
-            parseComplexEscapeSequence(s, buf);
+        {
+            if constexpr (parse_complex_escape_sequence)
+            {
+                parseComplexEscapeSequence(s, buf);
+            }
+            else
+            {
+                s.push_back(*buf.position());
+                ++buf.position();
+                if (!buf.eof())
```

Comment:
I guess there is no case. But I think it's more natural not to throw an exception in case when we don't want to parse escape sequences and just copy data as is, so the exception will be thrown in more appropriate place when we will parse all escape sequences. Like this:
```
avogar-dev :) desc format(TSV, $$Hello,World\$$)

DESCRIBE TABLE format(TSV, 'Hello,World\\')

Query id: 40f68190-f386-4a6d-8208-65a44aca2305

┌─name─┬─type─────────────┬─default_type─┬─default_expression─┬─comment─┬─codec_expression─┬─ttl_expression─┐
│ c1   │ Nullable(String) │              │                    │         │                  │                │
└──────┴──────────────────┴──────────────┴────────────────────┴─────────┴──────────────────┴────────────────┘

1 row in set. Elapsed: 0.005 sec.

avogar-dev :) select * from format(TSV, $$Hello,World\$$)

SELECT *
FROM format(TSV, 'Hello,World\\')

Query id: f37ee52d-25d7-4893-93c1-0621a9625e84


0 rows in set. Elapsed: 0.964 sec.

Received exception:
Code: 25. DB::Exception: Cannot parse escape sequence: While executing ParallelParsingBlockInputFormat. (CANNOT_PARSE_ESCAPE_SEQUENCE)

```